src/audio/host-legacy.c

```diff
@@ -234,35 +234,22 @@ void host_common_update(struct host_data *hd, struct comp_dev *dev, uint32_t byt
 {
 	struct comp_buffer __sparse_cache *source;
 	struct comp_buffer __sparse_cache *sink;
-	int ret;
 	bool update_mailbox = false;
 	bool send_ipc = false;
 
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
 		source = buffer_acquire(hd->dma_buffer);
 		sink = buffer_acquire(hd->local_buffer);
-		ret = dma_buffer_copy_from(source, sink, hd->process, bytes);
+		dma_buffer_copy_from(source, sink, hd->process, bytes);
 	} else {
 		source = buffer_acquire(hd->local_buffer);
 		sink = buffer_acquire(hd->dma_buffer);
-		ret = dma_buffer_copy_to(source, sink, hd->process, bytes);
+		dma_buffer_copy_to(source, sink, hd->process, bytes);
 	}
 
-	/* assert dma_buffer_copy succeed */
-	if (ret < 0)
-		comp_err(dev, "host_common_update() dma buffer copy failed, dir %d bytes %d avail %d free %d",
-			 dev->direction, bytes,
-			 audio_stream_get_avail_samples(&source->stream) *
-				audio_stream_frame_bytes(&source->stream),
-			 audio_stream_get_free_samples(&sink->stream) *
-				audio_stream_frame_bytes(&sink->stream));
-
 	buffer_release(sink);
 	buffer_release(source);
 
-	if (ret < 0)
-		return;
```

Comment:
if ret < 0 we will not update stream position for host, but now it is not correct

Reply:
there is no chance to get ret < 0 case, so I removed the return.